src/custom/CustomTooltip/customTooltip.tsx

```diff
@@ -27,6 +27,21 @@ function CustomTooltip({
   componentsProps = {},
   ...props
 }: CustomTooltipProps): JSX.Element {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const isTouchDevice = () => {
```

Comment:
i think `isTouchDevice` will run on every render. This is not good. Do you mind move it to `useMemo` .